buildpack.md

```diff
@@ -406,6 +409,14 @@ Note that buildpack IDs are expanded depth-first in left-to-right order.
 
 If a buildpack order entry within a group has the parameter `optional = true`, then a copy of the group without the entry MUST be repeated after the original group.
 
+#### Image Extensions
+
+Image extensions participate in the buildpack [detection](#detector) process, with the same `UID`, `GID`, and interface for `/bin/detect`. However:
+- Detection is optional for extensions, and they are assumed to pass detection when it is not present. A `/bin/detect` that exits with a 0 exit code passes detection, and fails otherwise.
+- Extensions MUST only output `provides` entries to the build plan. They MUST NOT output `requires`.
+- Extensions MUST all precede buildpacks in [order](#ordertoml-toml) definitions.
```

Comment:
https://github.com/buildpacks/samples/pull/113/files#r726206827 talks about creating a separate `[[order-ext]]` in builder.toml - would that translate into a separate file?